zerver/lib/user_groups.py

```diff
@@ -228,6 +228,80 @@ def access_user_group_for_setting(
     return user_group
 
 
+def access_user_groups_for_setting(
+    user_group_ids: List[int],
+    user_profile: UserProfile,
+    *,
+    setting_name: str,
+    permission_configuration: GroupPermissionSetting,
+) -> List[UserGroup]:
+    realm = user_profile.realm
+    user_groups = list(UserGroup.objects.filter(id__in=user_group_ids, realm=realm))
+    found_group_ids = [group.id for group in user_groups]
+
+    for group_id in user_group_ids:
+        if group_id not in found_group_ids:
+            raise JsonableError(_("Invalid user group ID: {group_id}").format(group_id=group_id))
+
+    for user_group in user_groups:
+        if permission_configuration.require_system_group and not user_group.is_system_group:
+            raise JsonableError(
+                _("'{setting_name}' must be a system user group.").format(setting_name=setting_name)
+            )
```

Comment:
I guess there's a bit of a question for this function whether we want the API to be enforcing that if `require_system_group`, then it must be a single system group? We obviously won't want that long term, but that might be the most coherent plan for flags like `allow_internet_group`.

Reply:
Certainly from an API format perspective, we want it to request a list of groups; but if it doesn't cause problems with reducing coverage or the like, it might be nice to not actually allow creating multiple groups. I suppose one option would be to have enforcement of "Exactly one group" that's disabled in a development environment so that we can merge this without having fixed everything to work with a set of multiple groups yet.

Reply:
I was actually thinking that having `require_system_group` as `True` would mean all the groups should be system groups. But that might actually not make sense atleast for the role-based system groups, as having a single user group should be enough.

Reply:
Yeah, I think keeping the behavior the same is probably simpler. In practice, a union of two role-based system groups will always have membership identical to just the larger of the two groups anyway.